docs/cudf/source/developer_guide/library_design.md

```diff
@@ -316,3 +316,171 @@ The pandas API also includes a number of helper objects, such as `GroupBy`, `Rol
 cuDF implements corresponding objects with the same APIs.
 Internally, these objects typically interact with cuDF objects at the Frame layer via composition.
 However, for performance reasons they frequently access internal attributes and methods of `Frame` and its subclasses.
+
+
+## Copy-on-write
+
```

Comment:
Somewhere in this section can we link to the pandas RFC for their implementation of copy-on-write to help motivate rationale/etc...

Reply:
Added it at the very end 👍